qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+        ]
+        return quantum_devices
+
+
+class AzureHelperFunctions:
+    """Helper functions for Azure Quantum API."""
+
+    def quantum_access_token(
+        self, client_id: str, client_secret: str, tenant_id: str, session: Any
+    ) -> str:
+        """Generates access token for Azure Quantum API."""
+        data = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "scope": "https://quantum.microsoft.com/.default",
+        }
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        url = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
+        r = session.post(url, headers=headers, data=data, verify=True)
+        quantum_access_token = r.json()["access_token"]
+
+        return quantum_access_token
+
+    def storage_access_token(
+        self, client_id: str, client_secret: str, tenant_id: str, session: Any
+    ) -> str:
+        """Generates access token for Azure Storage API."""
+        data = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "scope": "https://management.azure.com/.default",
+        }
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        url = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
+        r = session.post(url, headers=headers, data=data, verify=True)
+        storage_access_token = r.json()["access_token"]
+
+        return storage_access_token
+
+    @staticmethod
+    def device_manager(devices: dict) -> dict[str, dict[str, Any]]:
+        """Manages the data for all Azure Quantum devices."""
+        devices_dict = {}
+
+        for provider in devices["value"]:
+            for machine in provider["targets"]:
+                if machine["id"][:4] != "ionq":
+                    devices_dict[machine["id"]] = {
+                        "status": machine["currentAvailability"],
+                        "isAvailable": machine["currentAvailability"] == "Available",
+                        "nextAvailable": None,
+                        "availablilityCD": "",
+                        "averageQueueTime": machine["averageQueueTime"],
+                    }
+
+        return devices_dict
+
+    @staticmethod
+    def create_container(
+        token: str,
+        subscription_id: str,
+        resource_group_name: str,
+        storage_account_name: str,
+        session: Any,
+    ) -> list[str]:
+        """Create a new container for the job."""
+        storage_access_token = token
+
+        # NOTE: This step can be replaced with any code that qBraid has to generate job IDs,
+        # but even the official Azure documentation uses UUIDs for job IDs
+
+        job_id = str(uuid4())
+        container_name = f"job-{job_id}"  # name of the container to create
```

Comment:
Are you sure they allow you to assign your own job ID? what if it's not unique? I'm skeptical about this...

The container_name makes more sense to me, that's just where you plan to store the job data, kind of like an s3 bucket. So makes sense they would let you name that on your own.